prisma/schema.prisma

```diff
@@ -10,8 +10,39 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
-model Employee{
-  employeeId Int @id @default(autoincrement())
-  name String
-  salary Int @default(100000)
+model Employee {
+  employeeId   Int        @id @default(autoincrement())
+  name         String
+  salary       Int        @default(100000)
+  designation  Designation @default(Junior)
+  department   Department @relation(fields: [departmentId], references: [departmentId])
+  departmentId Int
+  projects ProjectAssignment[]
+}
+
+model Department {
+  departmentId    Int        @id @default(autoincrement())
+  departmentName String
+  Employee       Employee[]
+}
+
+model Project {
+  projectId Int @id @default(autoincrement())
+  projectName String
+  employees ProjectAssignment[]
+}
+
+model ProjectAssignment {
+  assignmentId Int @id @default(autoincrement())
+  employee Employee @relation(fields: [employeeId], references:[employeeId])
+  employeeId Int
+  project Project @relation(fields: [projectId], references:[projectId])
+  projectId Int
+}
+
+enum Designation {
+  Junior
+  Manager
+  Senior_Manager
```

Comment:
In our convention, we would write this as SeniorManager rather than Senior_Manager.